pg_backup_api/pg_backup_api/utils.py

```diff
@@ -76,3 +77,14 @@ def get_server_by_name(server_name):
         conf = barman.__config__.get_server(server)
         if server == server_name:
             return conf
+
+
+def parse_backup_id(server, backup_id):
+    if backup_id in ("latest", "last"):
+        backup_id = server.get_last_backup_id()
+    elif backup_id in ("oldest", "first"):
+        backup_id = server.get_first_backup_id()
+    elif args.backup_id in ("last-failed"):
+        backup_id = server.get_last_backup_id([BackupInfo.FAILED])
+
```

Comment:
This would also be a good place to resolve backup names to backup IDs (I know that's not a requirement in the issue description but I can see it being needed for similar reasons).

Reply:
sgtm, we'll do in different PR